front/app/components/admin/PostManager/ProposalsManager.tsx

```diff
@@ -0,0 +1,269 @@
+  MiddleColumnTop,
+  PreviewMode,
+  Sticky,
+  StyledExportMenu,
+  StyledInput,
+  ThreeColumns,
+  TopActionBar,
+} from '.';
+import ActionBar from './components/ActionBar';
+import FilterSidebar from './components/FilterSidebar';
+import PostTable from './components/PostTable';
+import InfoSidebar from './components/InfoSidebar';
+import InitiativesCount from './components/InitiativesCount';
+import FeedbackToggle from './components/TopLevelFilters/FeedbackToggle';
+const LazyPostPreview = lazy(
+  () => import('components/admin/PostManager/components/PostPreview')
+);
+import LazyStatusChangeModal from './components/StatusChangeModal/LazyStatusChangeModal';
+import Outlet from 'components/Outlet';
+
+type TFilterMenu = 'topics' | 'statuses';
+
+interface Props {
+  visibleFilterMenus: TFilterMenu[]; // cannot be empty.
+  defaultFilterMenu: TFilterMenu;
+}
+
+const ProposalsManager = ({ defaultFilterMenu, visibleFilterMenus }: Props) => {
+  const [selection, setSelection] = useState<Set<string>>(new Set());
+  const [activeFilterMenu, setActiveFilterMenu] =
+    useState<TFilterMenu>(defaultFilterMenu);
+  const [previewPostId, setPreviewPostId] = useState<string | null>(null);
+  const [previewMode, setPreviewMode] = useState<PreviewMode>('view');
+  const { data: initiativeStatuses } = useInitiativeStatuses();
+  const { data: initiativeTopics } = useTopics({ excludeCode: 'custom' });
+  const [queryParameters, setQueryParameters] = useState<IQueryParameters>({
+    pageSize: 10,
+    sort: 'new',
+  });
+  const { data: initiatives } = useInitiatives(queryParameters);
+
+  useEffect(() => {
+    setActiveFilterMenu((activeFilterMenu) => {
+      if (
+        !visibleFilterMenus.find((item) => item === activeFilterMenu) &&
+        visibleFilterMenus[0]
+      ) {
+        return visibleFilterMenus[0];
+      } else {
+        return activeFilterMenu;
+      }
+    });
+  }, [visibleFilterMenus]);
+
+  if (!initiatives) return null;
+
+  const resetSelection = () => {
+    setSelection(new Set());
+  };
+  // End selection management
+
+  // Filter menu
+  const handleChangeActiveFilterMenu = (activeFilterMenu: TFilterMenu) => {
+    setActiveFilterMenu(activeFilterMenu);
+  };
+  // End Filter menu
+
+  // Modal Preview
+  const openPreview = (postId: string) => {
+    setPreviewPostId(postId);
+    setPreviewMode('view');
+  };
+
+  const openPreviewEdit = () => {
+    const previewPostId = [...selection][0];
+    if (selection.size === 1 && previewPostId) {
+      setPreviewMode('edit');
+      setPreviewPostId(previewPostId);
+    }
+  };
+
+  const switchPreviewMode = () => {
+    if (previewMode === 'edit') {
+      setPreviewMode('view');
+    } else {
+      setPreviewMode('edit');
+    }
+  };
+
+  const closePreview = () => {
+    setPreviewPostId(null);
+    setPreviewMode('view');
+  };
+
+  const onChangeTopics = (topics: string[]) => {
+    setQueryParameters({ ...queryParameters, pageNumber: 1, topics });
+  };
```

Comment:
Wondering if we should bring these to `useInitiatives`/`useIdeas` like we had with `GetIdeas`/`GetInitiatives`, so you don't need to define `queryParameters` in the state of each component.

Reply:
The api hooks should be as simple as possible because they are used in many places but if you really need to reuse this behavior multiple times, you can always create a separate hook that wraps around `useInitiatives` and handles these methods. In this case, I don't think it makes a lot of sense because this is the only place we need these, right? So it's fine if they stay inside the component.